libpod/networking_linux.go

```diff
@@ -136,10 +137,30 @@ func (r *Runtime) setupRootlessNetNS(ctr *Container) (err error) {
 	if err := cmd.Start(); err != nil {
 		return errors.Wrapf(err, "failed to start process")
 	}
+	defer cmd.Process.Release()
 
 	b := make([]byte, 16)
-	if _, err := syncR.Read(b); err != nil {
-		return errors.Wrapf(err, "failed to read from sync pipe")
+	for {
+		if err := syncR.SetDeadline(time.Now().Add(1 * time.Second)); err != nil {
+			return errors.Wrapf(err, "error setting pipe timeout")
```

Comment:
Your error messages should mention that the problem is with slirp4netns hanging.

Reply:
I changed the error messages to mention slirp4netns